### PR TITLE
SWC-7185

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

**Comment by jay-hodgson** [comment]
https://github.com/user-attachments/assets/7e9b0474-a0ad-4f5e-9170-f3c2f167f857

File: src/main/java/org/sagebionetworks/web/client/widget/entity/browse/FilesBrowserViewImpl.java [hunk 1]
```diff
@@ -23,6 +26,8 @@ public interface FilesBrowserViewImplUiBinder
 
   private EntityTreeBrowser entityTreeBrowser;
   private Widget widget;
+  SynapseReactClientFullContextPropsProvider propsProvider;
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/browse/FilesBrowserViewImpl.java:29]: Consistency nit
```suggestion
  private SynapseReactClientFullContextPropsProvider propsProvider;
```

File: src/main/java/org/sagebionetworks/web/client/widget/entity/browse/FilesBrowserViewImpl.java [hunk 2]
```diff
@@ -53,15 +67,46 @@ public FilesBrowserViewImpl(
...
+    } else {
+      entityTreeBrowser.configure(entityId);
+    }
+  }
+
+  public void rerenderFileBrowser(String parentContainerId) {
+    // TODO: Should not need to remount, but currently the parentContainer is not shown if the existing react component is updated
+    // TODO: Bug: The tree is not always auto-expanding (to the path to the parent container Id)
+    reactFileBrowserContainer.clear();
+    ReactComponent componentContainer = new ReactComponent();
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/browse/FilesBrowserViewImpl.java:87]: You shouldn't need to create a new ReactComponent instance, you should just be able to create one instance and call `render` on it with the new ReactElement. But it's possible you tried that and it resulted in another bug.

**Review by nickgros — APPROVED** [review]
Looking forward to trying this!

File: src/main/java/org/sagebionetworks/web/client/widget/entity/browse/FilesBrowserViewImpl.java (hunk 2 above)
Comment (jay-hodgson) [reply src/main/java/org/sagebionetworks/web/client/widget/entity/browse/FilesBrowserViewImpl.java:87]: Yes, tried.  Noted a couple of bugs in TODO comments that would need to be addressed if we're moving forward with this solution

**Pull request merged**